### PR TITLE
Xaml UI

### PR DESCRIPTION
**Pull request by PaulRitter**
- [X] Compile XAML-Resources into IL-Code and weave it into assemblies
- [x] Construct Namespaces
- [x] Use Source-Generators to generate Properties based on Control:Name Property
- [x] Merge Namescopes (Example: SS14Window (with Namescope) and ConstructionMenu : SS14Window (with NameScope) -> SS14Window-Namescope gets overridden)

### CODE REVIEW

**Comment by PaulRitter** [comment]
so this is kinda done, review away

side note: sometimes msbuild fucks up and runs the task again (eg injects again) which will result into an error, just rebuild the project, that'll fix it

File: Robust.Client/Robust.Client.csproj [hunk 1]
```diff
@@ -45,4 +45,31 @@
...
+
+  <UsingTask TaskName="CompileRobustXamlTask" AssemblyFile="..\Robust.Client.Injectors\bin\$(Configuration)\net5.0\Robust.Client.Injectors.dll" />
+  <Target Name="CompileRobustXaml" AfterTargets="AfterCompile">
+    <PropertyGroup>
+      <RobustXamlReferencesTemporaryFilePath Condition="'$(RobustXamlReferencesTemporaryFilePath)' == ''">$(IntermediateOutputPath)XAML/references</RobustXamlReferencesTemporaryFilePath>
+      <RobustXamlOriginalCopyFilePath Condition="'$(RobustXamlOriginalCopyFilePath)' == ''">$(IntermediateOutputPath)XAML/original.dll</RobustXamlOriginalCopyFilePath>
+    </PropertyGroup>
+    <WriteLinesToFile File="$(RobustXamlReferencesTemporaryFilePath)" Lines="@(ReferencePathWithRefAssemblies)" Overwrite="true" />
+    <CompileRobustXamlTask AssemblyFile="@(IntermediateAssembly)" ReferencesFilePath="$(RobustXamlReferencesTemporaryFilePath)" OriginalCopyPath="$(RobustXamlOriginalCopyFilePath)" ProjectDirectory="$(MSBuildProjectDirectory)" AssemblyOriginatorKeyFile="$(AssemblyOriginatorKeyFile)" SignAssembly="$(SignAssembly)" DelaySign="$(DelaySign)" />
+  </Target>
```
Comment (PJB3005) [inline Robust.Client/Robust.Client.csproj]: Shouldn't this be in a .targets file?

File: Robust.Client/UserInterface/CustomControls/SS14Window.xaml [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+            <controls:HBoxContainer>
+                <controls:MarginContainer MarginLeftOverride="5" SizeFlagsHorizontal="FillExpand">
+                    <controls:Label Name="TitleLabel" StyleIdentifier="foo" ClipText="True" Text="Exemplary Window Title Here" VAlign="Center" StyleClasses="windowTitle"/>
+                </controls:MarginContainer>
+                <controls:TextureButton Name="CloseButton" StyleClasses="windowCloseButton" SizeFlagsVertical="ShrinkCenter"></controls:TextureButton>
+            </controls:HBoxContainer>
+        </controls:PanelContainer>
+        <controls:MarginContainer Name="ContentsContainer" MarginBottomOverride="10" MarginLeftOverride="10" MarginRightOverride="10" MarginTopOverride="10" RectClipContent="True" SizeFlagsVertical="FillExpand" />
+    </controls:VBoxContainer>
+</customControls:SS14Window>
```
Comment (PJB3005) [inline Robust.Client/UserInterface/CustomControls/SS14Window.xaml]: Seriously, make some shorter XML namespaces here. It could be so much more readable.

Comment (PaulRitter) [reply Robust.Client/UserInterface/CustomControls/SS14Window.xaml]: the autocomplete is to blame here. luckily you can easily just do the funny ctrl + r + r to rename the namespace. will do

File: Robust.Client/Robust.Client.csproj (hunk 1 above)
Comment (PaulRitter) [reply Robust.Client/Robust.Client.csproj]: sadly i cannot put it in a targets file and use in in Content.Client due to it then having the wrong paths :/ If you have an idea how to solve this please do tell.
To elaborate with an example `..\Robust.Client.XAMLUI\Robust.Client.XAMLUI.csproj` gets transformed into
- Robust.Client: `SS14\RobustToolbox\Robust.Client.XAMLUI\Robust.Client.XAMLUI.csproj`
- Content.Client: `SS14\Robust.Client.XAMLUI\Robust.Client.XAMLUI.csproj`

Comment (PJB3005) [reply Robust.Client/Robust.Client.csproj]: Use `$(MSBuildThisFileDirectory)`

**Comment by PJB3005** [comment]
What could possibly go wrong?

**Pull request merged**